src/enclave/enclave_init.c

```diff
@@ -89,8 +91,60 @@ static void init_wireguard()
         wgu_list_devices();
 }
 
-static int startmain(void* args)
+static long _lkl_syscall_wrapper(long no, long* params)
+{
+    //sgxlkl_warn("syscall begin: no=%u\n", no);
+    long ret = lkl_syscall(no, params);
+    //sgxlkl_warn("syscall end: ret=%u\n", ret);
+    return ret;
+}
+
+static void _enter_user_space(
+    int argc,
+    char** argv,
+    void* stack,
+    size_t num_ethreads,
+    struct timespec clock_res[4])
+{
+    extern void* __oe_get_isolated_image_entry_point(void);
+    extern const void* __oe_get_isolated_image_base();
+    typedef int (*sgxlkl_user_enter_proc_t)(void* args, size_t size);
+    sgxlkl_userargs_t args;
+    sgxlkl_user_enter_proc_t proc;
+
+    memset(&args, 0, sizeof(args));
+
+    if (!(proc = __oe_get_isolated_image_entry_point()))
+        sgxlkl_fail("failed to obtain user space entry point");
+
+    args.ua_lkl_syscall = _lkl_syscall_wrapper;
```

Comment:
See above comment about dead code.
```suggestion
    args.ua_lkl_syscall = lkl_syscall;
```